activiti-spring-conformance-tests/activiti-spring-conformance-signals/src/test/java/org/activiti/spring/conformance/signals/SignalThrowCatchTest.java

```diff
@@ -0,0 +1,288 @@
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.activiti.spring.conformance.signals.SignalsRuntimeTestConfiguration.collectedEvents;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.tuple;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.NONE)
+public class SignalThrowCatchTest {
+
+    @Autowired
+    private ProcessRuntime processRuntime;
+
+    @Autowired
+    private ProcessAdminRuntime processAdminRuntime;
+
+    @Autowired
+    private SecurityUtil securityUtil;
+
+    @Before
+    public void cleanUp() {
+        collectedEvents.clear();
+    }
+
+    @Test
+    public void testProcessWithThrowSignal() {
+        securityUtil.logInAs("user1");
+
+        String processInstaneId = startThrowSignalProcess();
+
+        assertThat(collectedEvents)
+                .extracting(RuntimeEvent::getEventType)
+                .containsExactly(
+                        ProcessRuntimeEvent.ProcessEvents.PROCESS_CREATED,
+                        ProcessRuntimeEvent.ProcessEvents.PROCESS_STARTED,
+                        BPMNActivityEvent.ActivityEvents.ACTIVITY_STARTED,
+                        BPMNActivityEvent.ActivityEvents.ACTIVITY_COMPLETED,
+                        BPMNSequenceFlowTakenEvent.SequenceFlowEvents.SEQUENCE_FLOW_TAKEN,
+                        BPMNActivityEvent.ActivityEvents.ACTIVITY_STARTED,
+                        BPMNActivityEvent.ActivityEvents.ACTIVITY_COMPLETED,
+                        BPMNSequenceFlowTakenEvent.SequenceFlowEvents.SEQUENCE_FLOW_TAKEN,
+                        BPMNActivityEvent.ActivityEvents.ACTIVITY_STARTED,
+                        BPMNActivityEvent.ActivityEvents.ACTIVITY_COMPLETED,
+                        ProcessRuntimeEvent.ProcessEvents.PROCESS_COMPLETED
+                );
+
+        assertThat(collectedEvents)
+                .filteredOn(event -> event.getEventType().equals(BPMNActivityEvent.ActivityEvents.ACTIVITY_STARTED))
+                .filteredOn(event -> ((BPMNActivity) event.getEntity()).getActivityType().equals("throwEvent"))
+                .extracting(event -> ((BPMNActivity) event.getEntity()).getActivityType(),
+                            event -> ((BPMNActivity) event.getEntity()).getProcessInstanceId())
+                .contains(
+                        tuple("throwEvent",
+                              processInstaneId));
+
+        collectedEvents.clear();
+    }
+
+    @Test
+    public void testProcessWithIntermediateCatchEventSignal() {
+        securityUtil.logInAs("user1");
+
+        String processInstanceId = startIntermediateCatchEventSignalProcess();
+
+        SignalPayload signalPayload = ProcessPayloadBuilder.signal()
+                .withName("Test")
+                .withVariable("signal-variable",
+                              "test")
+                .build();
+        processRuntime.signal(signalPayload);
```

Comment:
@CTI777 this test will be more clear if we split the operations and the assertions in half.. 
1) Start and the assertions
   1.1) We can check that the last event here is the BPMN element that is going to catch the signal. 
2) Signal and the assertions